### PR TITLE
[Backport stable/8.3] ci: use infra self-hosted runners (INFRA-671)

### PR DESCRIPTION
**Pull request by clementnero**
## Description

Related to https://github.com/camunda/team-infrastructure/issues/671.

Custom backport PR for:
- https://github.com/camunda/camunda/pull/22033
- https://github.com/camunda/camunda/pull/22450
- https://github.com/camunda/camunda/pull/22609
- https://github.com/camunda/camunda/pull/22807

Strace-related tests are not executed on CI for `stable/8.3`, therefore no need to extract them into a separate job.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Comment by clementnero** [comment]
Strace tests are excluded from CI in stable/8.3, I will reflect that (just made some tests for now).

**Review by cmur2 — COMMENTED** [review]
I'm fine here but since branches before 8.6 concern only Zeebe and I'm not aware of the history, please also get a review from them.

File: .github/workflows/zeebe-ci.yml [hunk 1]
```diff
@@ -128,12 +131,10 @@ jobs:
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
   unit-tests:
     name: Unit tests
-    runs-on: [ self-hosted, linux, amd64, "16" ]
+    runs-on: gcp-perf-core-16-default
     timeout-minutes: 30
     steps:
       - uses: actions/checkout@v4
-      - name: Install strace tests
-        run: sudo apt-get -qq update && sudo apt-get install -y strace
```
Comment (npepinpe) [inline .github/workflows/zeebe-ci.yml:136]: :question: So we backported already the other changes?

File: .github/workflows/zeebe-ci.yml [hunk 2]
```diff
@@ -156,8 +157,6 @@ jobs:
           -P skip-random-tests,parallel-tests,extract-flaky-tests
           verify
           | tee "${BUILD_OUTPUT_FILE_PATH}"
-      - name: Normalize artifact name
-        run: echo "ARTIFACT_NAME=$(echo ${{ matrix.project }} | sed 's/\//-/g')" >> $GITHUB_ENV
```
Comment (npepinpe) [inline .github/workflows/zeebe-ci.yml:160]: :question: Why is this required?

**Review by npepinpe — APPROVED** [review]
Nothing major, just some questions, but approving as I don't think it should block.

File: .github/workflows/zeebe-ci.yml (hunk 1 above)
Comment (clementnero) [reply .github/workflows/zeebe-ci.yml:136]: Strace tests are disabled in `stable/8.3`, see https://github.com/camunda/camunda/blob/stable/8.3/snapshot/src/test/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksumTest.java#L133` annotation.
This step is not necessary.

File: .github/workflows/zeebe-ci.yml (hunk 2 above)
Comment (clementnero) [reply .github/workflows/zeebe-ci.yml:160]: This step is a leftover, I removed it as such. The `ARTIFACT_NAME ` env will be always empty (there is no matrix defined, `${{ matrix.project }}`).
